Modules/_io/stringio.c

```diff
@@ -943,26 +971,67 @@ stringio_setstate(stringio *self, PyObject *state)
 static PyObject *
 stringio_closed(stringio *self, void *context)
```

Comment:
I think these functions will be easier to read and more consistent with the other parts of the file if you structure it like.

```
static PyObject *
stringio_closed_impl(stringio *self, void *context)
{
    CHECK_INITIALIZED(self);
    return PyBool_FromLong(self->closed);
}


static PyObject *
stringio_closed(stringio *self, void *context)
{
    PyObject *result;
    Py_BEGIN_CRITICAL_SECTION(self);
    result = stringio_closed_impl(self, context);
    Py_END_CRITICAL_SECTION();
    return result;
}
```